### PR TITLE
[DOCS] Add note about `meta` field behavior

### LINKED ISSUE #1822: [DOCS] Update exception item - meta field
## Description 

There is [an sdh](https://github.com/elastic/sdh-security-team/issues/352) where the customer was using the detection rules `meta` field to store arbitrary user data. This field is meant to be hands off from a user perspective. We should note in the docs that this field should not be used to store arbitrary user data as it is overwritten every time a rule is updated.

## Acceptance Test Criteria

As a user, when I check the [Update exception item API docs](https://www.elastic.co/guide/en/security/current/exceptions-api-update-item.html), I should see a note that the meta field will be overwritten on update and should not be used to store arbitrary data.

### PR DESCRIPTION
**Pull request by joepeeples**
Resolves #1822.

Previews:
* [Create rule](https://security-docs_2000.docs-preview.app.elstc.co/guide/en/security/master/rules-api-create.html#opt-fields-all)
* [Update rule](https://security-docs_2000.docs-preview.app.elstc.co/guide/en/security/master/rules-api-update.html#_optional_fields_for_all_rule_types)

### CODE REVIEW

File: docs/detections/api/rules/rules-api-create.asciidoc [hunk 1]
```diff
@@ -285,7 +285,9 @@ means the rule runs every hour. Defaults to `5m` (5 minutes).
 |max_signals |Integer |Maximum number of alerts the rule can create during a
 single execution. Defaults to `100`.
 
-|meta |Object |Placeholder for metadata about the rule.
+|meta |Object a|Placeholder for metadata about the rule.
+
+*NOTE*: This field is overwritten when you save any changes to the rule's settings.
```
Comment (nastasha-solomon) [inline docs/detections/api/rules/rules-api-create.asciidoc:290]:
```suggestion
*NOTE*: This field is overwritten when you save changes to the rule's settings.
```

File: docs/detections/api/rules/rules-api-update.asciidoc [hunk 2]
```diff
@@ -205,7 +205,9 @@ means the rule runs every hour. Defaults to `5m` (5 minutes).
 |max_signals |Integer |Maximum number of alerts the rule can create during a
 single execution. Defaults to `100`.
 
-|meta |Object |Placeholder for metadata about the rule.
+|meta |Object a|Placeholder for metadata about the rule.
+
+*NOTE*: This field is overwritten when you save any changes to the rule's settings.
```
Comment (nastasha-solomon) [inline docs/detections/api/rules/rules-api-update.asciidoc:210]:
```suggestion
*NOTE*: This field is overwritten when you save changes to the rule's settings.
```

**Review by nastasha-solomon — APPROVED** [review]
Left two minor edits that you can take or leave. Otherwise, LGTM!

**Review by spong — APPROVED** [review]
LGTM -- thanks @joepeeples! 🙂

**Review by jmikell821 — APPROVED** [review]
LGTM!

**Pull request merged**